### PR TITLE
Fixed return constant value cache on 32 bit VMs

### PR DESCRIPTION
**Pull request by rartemev**
Fixed DevDiv_491206.
The problem was about `MergedReturns` optimization. Since we have only `ptrdiff_t` int cache for return constants so we cannot optimize long ones on 32 bit VMs. The PR fixes this adding support of long values too.

### CODE REVIEW

**Comment by rartemev** [comment]
cc: @dotnet/jit-contrib

**Comment by rartemev** [comment]
@BruceForstall fixed

**Comment by rartemev** [comment]
@dotnet-bot test OSX10.12 x64 Checked Build and Test
@dotnet-bot test Ubuntu x64 Checked Build and Test
@dotnet-bot test Windows_NT x64 Debug Build and Test
@dotnet-bot test Windows_NT x86 Checked Build and Test

**Comment by rartemev** [comment]
@dotnet-bot test Windows_NT x64 Release Priority 1 Build and Test

File: src/jit/gentree.h [hunk 1]
```diff
@@ -2543,6 +2543,7 @@ struct GenTreeIntConCommon : public GenTree
     inline void SetLngValue(INT64 val);
     inline ssize_t IconValue();
     inline void SetIconValue(ssize_t val);
+    inline INT64 IntergalValue();
```
Comment (BruceForstall) [inline src/jit/gentree.h]: typo: IntergralValue => IntegralValue

**Comment by rartemev** [comment]
@dotnet-bot test Windows_NT x64 Formatting

**Pull request merged**